web/app/js/components/Deployments.jsx

```diff
@@ -78,49 +78,34 @@ export default class Deployments extends React.Component {
   }
 
   loadTimeseriesFromServer(meshDeployMetrics, combinedMetrics) {
+    // fetch only the timeseries for the 3 deployments we display at the top of the page
     let limitSparklineData = _.size(meshDeployMetrics) > maxTsToFetch;
     let rollupPath = `${this.props.pathPrefix}/api/metrics?window=${this.state.metricsWindow}`;
     let timeseriesPath = `${rollupPath}&timeseries=true`;
+    let leastHealthyDeployments = this.getLeastHealthyDeployments(meshDeployMetrics);
 
-    let updatedState = {
-      metrics: combinedMetrics,
-      limitSparklineData: limitSparklineData,
-      loaded: true,
-      pendingRequests: false,
-      error: ''
-    };
+    let tsPromises = _.map(leastHealthyDeployments, dep => {
+      let tsPathForDeploy = `${timeseriesPath}&target_deploy=${dep.name}`;
```

Comment:
yep! the reason I didn't before was that I was originally trying to have urlsForResource also be a reference for the group by as well and I wasn't sure I wanted to encode every combination of url params and group bys in that const. But I agree it'd be nice to stop constructing urls everywhere (also this is a deployment anyway, so that grouping works), so I'll just move them all over.